fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanEventApiJsonValidator.java

```diff
@@ -518,6 +518,26 @@ public void validateLoanChargeRefundTransaction(final String json) {
         throwExceptionIfValidationWarningsExist(dataValidationErrors);
     }
 
+    public void validateSetFlagToLoan(final String json) {
+        if (StringUtils.isBlank(json)) {
+            return;
+        }
+        Set<String> transactionParameters = new HashSet<>(Arrays.asList(LoanApiConstants.FRAUD_FLAG_NAME));
+
+        final Type typeOfMap = new TypeToken<Map<String, Object>>() {}.getType();
+        this.fromApiJsonHelper.checkForUnsupportedParameters(typeOfMap, json, transactionParameters);
+
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors).resource("loan.flags");
```

Comment:
Extract the "loan.flags" into a constant pls.

Reply:
Done!